SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -219,32 +219,45 @@ class run_test {
     });
     queue.wait_and_throw();
 
-    bool passed = true;
+    if (!are_bitwise_equal(result[0], base_value)) {
+      fail_test(0, result[0], base_value, data_type);
+    }
 
     DataT expected_value = base_value;
-    for (size_t i = 0; i < result.size(); ++i) {
-      // std::isnan() couldn't be called for integral types because it call is
-      // ambiguous GitHub issue for that case:
-      // https://github.com/microsoft/STL/issues/519
-      if (!are_bitwise_equal(result[i], expected_value)) {
-        if constexpr (type_traits::is_sycl_floating_point_v<DataT>) {
-          if (std::isnan(result[i]) &&
-              (std::isnan(step_value) || std::isnan(base_value))) {
-            continue;
+    if constexpr (BaseVal::value == init_val::nan ||
+                  Step::value == init_val::nan) {
+      for (size_t i = 1; i < result.size(); ++i) {
+        if (!std::isnan(result[i])) {
+          if (!are_bitwise_equal(result[i], expected_value)) {
+            fail_test(i, result[i], expected_value, data_type);
           }
         }
-        passed = false;
-
-        const auto description =
-            FillCtorTestDescription<DataT, NumElems, TestCaseT, BaseVal::value,
-                                    Step::value>(i, result[i - 1] + step_value,
-                                                 expected_value, data_type);
-        log::fail(description);
+        expected_value += step_value;
+      }
+    } else {
+      for (size_t i = 1; i < result.size(); ++i) {
+        if (!are_bitwise_equal(result[i], expected_value)) {
+          fail_test(i, result[i], expected_value, data_type);
+        }
       }
       expected_value += step_value;
     }
 
-    return passed;
+    return m_passed;
+  }
+
+private:
+  bool m_passed = true;
+
+  void fail_test(size_t step, DataT retrieved, DataT expected,
```

Comment:
```suggestion
  void fail_test(size_t index, DataT retrieved, DataT expected,
```
The `base` and `step` has specific meaning for fill constructor.

Reply:
this comment was applied [here](https://github.com/intel/llvm-test-suite/pull/598/commits/04d6bab9c3fb63aeef04dbeea45f802d05edce82#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4L258-L259)